x-pack/plugins/ml/public/file_datavisualizer/components/import_view/importer/sst_importer.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { Importer } from './importer';
+
+export class SstImporter extends Importer {
+  constructor(results) {
+    super(results);
+
+    this.format = results.format;
+    this.multilineStartPattern = results.multiline_start_pattern;
+    this.grokPattern = results.grok_pattern;
+  }
+
+  // convert the semi structured text string into an array of lines
+  // by looking over each char, looking for newlines.
+  // if one is found, check the next line to see if it starts with the
+  // multiline_start_pattern regex
+  // if it does, it is a legitimate end of line and can be pushed into the list,
+  // if not, it must be a new line char inside a field value, so keep looking.
+  async read(text) {
+    console.log('read sst file');
+    console.time('read sst file');
+
+    try {
+      const data = [];
+
+      let line = '';
+      let tempLine = '';
```

Comment:
It might be clearer to rename `line` to `message` and and `tempLine` to `line`.  As it stands `line` can have more than one line.